streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorNodeTest.java

```diff
@@ -141,4 +154,76 @@ public void testMetrics() {
                 groupName, context.taskId().toString(), node.name())));
     }
 
+    @Test(expected = StreamsException.class)
+    public void testTopologyLevelClassCastException() {
+        final Properties props = new Properties();
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "test");
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234");
+        // Serdes configuration is missing (default will be used which don't match the DSL below), which will trigger the new exception
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder.<String, String>stream("streams-plaintext-input")
+            .flatMapValues(value -> {
+                return Arrays.asList("");
+            });
+        final Topology topology = builder.build();
+
+        final TopologyTestDriver testDriver = new TopologyTestDriver(topology, props);
+        final ConsumerRecordFactory<String, String> factory = new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer());
+        final ConsumerRecord<byte[], byte[]> consumerRecord = factory.create("streams-plaintext-input", "a-key", "a value");
+
+        try {
+            testDriver.pipeInput(consumerRecord);
+        } catch (final StreamsException s) {
+            final String msg = s.getMessage();
+            assertTrue("Error about class cast with serdes", msg.contains("ClassCastException"));
+            assertTrue("Error about class cast with serdes", msg.contains("serdes"));
+            throw s;
+        }
+    }
+
+    @Test
+    public void testTopologyLevelClassCastExceptionCorrect() {
```

Comment:
Do we need this test?

Reply:
Yeah I was thinking about this, I don't think we do. I'll remove it.